includes/sanitizers/class-amp-tag-and-attribute-sanitizer.php

```diff
@@ -1445,6 +1450,56 @@ private function is_valid_layout( $tag_spec, $node ) {
 		return true;
 	}
 
+	/**
+	 * Whether the node is inside a mustache template.
+	 *
+	 * @since 1.5.3
+	 *
+	 * @param DOMElement $node The node to examine.
+	 * @return bool Whether the node is inside a valid mustache template.
+	 */
+	private function is_inside_mustache_template( DOMElement $node ) {
+		if ( ! empty( $this->open_elements['template'] ) ) {
+			while ( $node->parentNode instanceof DOMElement ) {
+				$node = $node->parentNode;
+				if ( Tag::TEMPLATE === $node->nodeName && 'amp-mustache' === $node->getAttribute( 'type' ) ) {
+					return true;
+				}
+			}
+		} elseif ( ! empty( $this->open_elements['script'] ) ) {
+			while ( $node->parentNode instanceof DOMElement ) {
+				$node = $node->parentNode;
+				if ( Tag::SCRIPT === $node->nodeName && 'amp-mustache' === $node->getAttribute( 'template' ) && 'text/plain' === $node->getAttribute( 'type' ) ) {
+					return true;
+				}
+			}
+		}
+		return false;
+	}
+
+	/**
+	 * Whether the node has a layout attribute with variable syntax, like {{foo}}.
+	 *
+	 * This is important for whether to validate the layout of the node.
+	 * Similar to the validation logic in the AMP validator.
+	 *
+	 * @see https://github.com/ampproject/amphtml/blob/c083d2c6120a251dcc9b0beb33c0336c7d3ca5a8/validator/engine/validator.js#L4038-L4054
+	 *
+	 * @since 1.5.3
+	 *
+	 * @param DOMElement $node The node to examine.
+	 * @return bool Whether the node has a layout attribute with variable syntax.
+	 */
+	private function has_layout_attribute_with_mustache_variable( DOMElement $node ) {
+		foreach ( [ 'layout', 'width', 'height', 'sizes', 'heights' ] as $attribute ) {
+			if ( preg_match( '#{{.*}}#', $node->getAttribute( $attribute ) ) ) {
```

Comment:
Might as well make this match what is in the style sanitizer:

```suggestion
			if ( preg_match( '/{{[^}]+?}}/', $node->getAttribute( $attribute ) ) ) {
```

Reply:
OK, sounds good.